s2i/builder.go

```diff
@@ -76,6 +76,13 @@ func (b *Builder) Build(ctx context.Context, f fn.Function) (err error) {
 	cfg.RuntimeImagePullPolicy = api.DefaultRuntimeImagePullPolicy
 	cfg.DockerConfig = s2idocker.GetDefaultDockerConfig()
 
+	// Excludes
+	// Do not include .git, .env, .func or any language-specific cache directories
+	// (node_modules, etc) in the tar file sent to the builder, as this both
+	// bloats the build process and can cause unexpected errors in the resultant
+	// Function.
+	cfg.ExcludeRegExp = "(^|/)\\.git|\\.env|\\.func|node_modules(/|$)"
```

Comment:
Why is `(^|/)` and `(^|/)` is required?

Reply:
Don't we also exclude the `func.yaml` file from the image since we do it in our buildpacks?

Reply:
The first one is such that it catches them when the file is nested, but not a _part_ of a filename, the second question is that the func.yaml was requested to be included in the image by someone for introspection, becase it represents actual metadata about the Function which is pertinent.  Note how `.func` is not included, because that is expressly transient local-only metadata.  I would lean toward _not_ including func.yaml in the build, because I don't want any third-party systems to start relying on the exact format of func.yaml, but instead use `func` directly, or the client library where possible.  Perhaps we can pop the question to the community and ask if anyone has a good reason to keep it in